src/Widgets/Views/MainView.vala

```diff
@@ -109,19 +113,38 @@ public class Bluetooth.Widgets.MainView : Gtk.Box {
     }
 
     private void update_devices_box_visible () {
-        devices_box.no_show_all = (devices_box.get_children ().length () <= 1);
-        devices_box.visible = !devices_box.no_show_all;
+        bool has_visible_device = false;
+        foreach (var device in devices.values) {
+            if (!device.no_show_all) {
```

Comment:
The visible property was actually returning false for me, I guess visible returns false if it still hasen't been shown on screen :shrug: